index.js

```diff
@@ -173,6 +173,10 @@ function readStream (stream, encoding, length, limit, callback) {
     }))
   }
 
+  if (stream.hasOwnProperty('readable') && !stream.readable) {
```

Comment:
Why the has own property? Streams can be subclassed.

Reply:
Sorry It maybe unreliable, I add `hasOwnProperty ` just for the situation where the stream is not a instance of http.IncomingMessage. For example, it have not `readable` property when a instance of `EventEmitter`

Reply:
how about `if (typeof stream.readable !== 'undefined' && !stream.readable)`? stream have so many way implemented，testing readable of each of those could be complicated, should we just care about the `http.IncomingMessage` situation?

Reply:
Wouldn't that pretty must be identical to `stream.readable === false` ? Or would that not catch a certain case (it's hard to say since the `readable` property doesn't seem to be documented anywhere in Node.js documentation -- how reliable is using this property?).

> should we just care about the http.IncomingMessage situation?

The main two uses cases for this module is reading the body from a request on the server and reading the body from a response in the client. Theses are the use cases that need to work. Ideally it works for generic readable streams too, since we've always supported this, because sometimes you need to pipe the request/response though zlib or similar to decompress it.

Reply:
Sounds convincing, I can make a try finding some reliable way check it.